File: qmessentials-services-dotnet/qmessentials-services-dotnet/Models/Observation.cs
using System;

namespace QMEssentials.Models
{

    public record Observation
    {
        public Observation(string itemId, string metricId, string lotId, string productId, double value,
            DateTime createdDate)
        {
            ItemId = itemId;
            MetricId = metricId;
            LotId = lotId;
            ProductId = productId;
            Value = value;
            CreatedDate = createdDate;
        }

        public string ItemId { get; }
        public string MetricId { get; }
        public string LotId { get;  }
        public string ProductId { get; }
        public double Value { get; }
        public DateTime CreatedDate { get; }
    }

}

File: qmessentials-observation-service/models/observation.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Observation struct {
	ObservationID      uuid.UUID `bson:"observationId" json:"observationId"`
	MetricID           int       `bson:"metricId" json:"metricId"`
	ItemID             string    `bson:"itemId" json:"itemId"`
	LotID              string    `bson:"lotId" json:"lotId"`
	ProductID          string    `bson:"productId" json:"productId"`
	LotSequenceNumber  int       `bson:"lotSequenceNumber" json:"lotSequenceNumber"`
	ItemSequenceNumber int       `bson:"itemSequenceNumber" json:"itemSequenceNumber"`
	Modifiers          []string  `bson:"modifiers" json:"modifiers"`
	Values             []float64 `bson:"values" json:"values"`
	CreatedDate        time.Time `bson:"createdDate" json:"createdDate"`
}
